cmd/registry/controller/local-tasks.go

```diff
@@ -38,59 +27,21 @@ type ExecCommandTask struct {
 	TaskID string
 }
 
-func getCommand(ctx context.Context, action []string) (*cobra.Command, error) {
-	if len(action) == 0 {
-		return nil, fmt.Errorf("empty action string")
-	}
-
-	switch action[0] {
-	case "annotate":
-		return annotate.Command(ctx), nil
-	case "compute":
-		return compute.Command(ctx), nil
-	case "delete":
-		return delete.Command(ctx), nil
-	case "export":
-		return export.Command(ctx), nil
-	case "get":
-		return get.Command(ctx), nil
-	case "index":
-		return index.Command(ctx), nil
-	case "label":
-		return label.Command(ctx), nil
-	case "list":
-		return list.Command(ctx), nil
-	case "search":
-		return search.Command(ctx), nil
-	case "upload":
-		return upload.Command(ctx), nil
-	case "vocabulary":
-		return vocabulary.Command(ctx), nil
-	case "resolve":
-		return nil, fmt.Errorf("cannot use command 'resolve' in action")
-	default:
-		return nil, fmt.Errorf("action '%s' is not supported", strings.Join(action, " "))
-	}
-}
-
 func (task *ExecCommandTask) String() string {
 	return "Execute command: " + task.Action
 }
 
 func (task *ExecCommandTask) Run(ctx context.Context) error {
-	args := strings.Fields(task.Action)
-	cmd, err := getCommand(ctx, args)
+	if strings.HasPrefix(task.Action, "resolve") {
+		return fmt.Errorf("'resolve' not allowed in action, cannot invoke %q", task.Action)
+	}
+	cmd := exec.Command("registry", strings.Fields(task.Action)...)
+	output, err := cmd.CombinedOutput()
+
 	if err != nil {
 		return err
 	}
-	// Set args if any
-	if len(args) > 1 {
-		cmd.SetArgs(args[1:])
-	}
-	if err := cmd.Execute(); err != nil {
-		return fmt.Errorf("error executing action %s: %s", task.Action, err)
-	}
+	log.Printf("Finished executing taskId %s with action: %s\n Output: %s", task.TaskID, task.Action, output)
```

Comment:
Here we can use `cmd.Stdout, cmd.Stderr = os.Stdout, os.Stderr` to hook the process directly into the correct outputs. The logs for each action will be interleaved without any way to identify them separately right now though.

No action necessary if you prefer the current output until we have unique identifiers attached to command logs.

Reply:
Noted. I am planning to take care of the logging change in a separate PR.